### ISSUE TITLE
license grant talking about "MIT X11" wrongly detected as MIT (not also alternatively X11)

### ISSUE THREAD

**Issue by jonassmedegaard** [nexB/scancode-toolkit#2634]
The test file `src/licensedcode/data/rules/mit_40.RULE` is detected as the MIT License, where in fact it is ambiguous: It can either be "the" MIT/X11 License, or it can be the X11-not-Expat variant of MIT licenses.
An example of the latter meaning of the term is at https://github.com/wayland-project/wayland/blob/main/CONTRIBUTING.md

**Comment by pombredanne** [comment]
This one is not straightforward to solve
This was originally found in .Net and mono with this wording. 
Yet you are right that this is ambiguous as witnessed in the case of your wayland finding.

It does not feel right to make this a choice of license... rather I would:
1. lower the relevance of the rule https://github.com/nexB/scancode-toolkit/blob/b7d070fc788f4c233df1f31d735e1ac5b3aa2d29/src/licensedcode/data/rules/mit_40.yml#L3 as this is not 100% .. but more like 90%.
2. add notes to the rule explaining the ambiguity in notes with pointers to various cases
3. add new rules specifically for wayland and related that would take care of deailing with this ambiguity

**Comment by jonassmedegaard** [comment]
I fully agree that there is no choice possibe - instead this issue is about certainty being too high.

I would say that certainty is 50% - by the reasoning that popularity of one interpretation over the other is irrelevant for a concrete case.

**Comment by jonassmedegaard** [comment]
What I mean is that it is irrelevant if 90% of code authors thinks that "MIT X11" is equivalent to SPDX MIT.
For each concrete case the only thing that matters is what the author meant - and based on the presented evidence (the text itself) there is a 50% chance that the author meant equivalence with SPDX X11.

**Comment by jonassmedegaard** [comment]
...otherwise, should scancode-toolkit interpret the lack of any license statement as 60% chance of MIT by the reasoning that statistically MIT is popular and statistically authors forget to license their work... :-P
